dCommon/GeneralUtils.h

```diff
@@ -234,18 +234,30 @@ namespace GeneralUtils {
 		return T();
 	}
 
-// on Windows we need to undef these or else they conflict with our numeric limits calls
-// DEVELOPERS DEVELOPERS DEVELOPERS DEVELOPERS DEVELOPERS DEVELOPERS DEVELOPERS DEVELOPERS
-#ifdef _WIN32
-#undef min
-#undef max
-#endif
+	/**
+	 * Casts the value of an enum entry to its underlying type
+	 * @param entry Enum entry to cast
+	 * @returns The enum entry's value in its underlying type
+	*/
+	template <typename T>
+	inline constexpr typename std::underlying_type<T>::type CastUnderlyingType(const T& entry) {
+		static_assert(std::is_enum<T>::value, "Not an enum");
+
+		return static_cast<typename std::underlying_type<T>::type>(entry);
```

Comment:
can be shortened to underlying_type_t<T>

Would also be nice to have a better template name aside from the general T